app/2fa/server/loginHandler.js

```diff
@@ -22,3 +25,57 @@ callbacks.add('onValidateLogin', (login) => {
 
 	return login;
 }, callbacks.priority.MEDIUM, '2fa');
+
+const recreateError = (errorDoc) => {
+	let error;
+
+	if (errorDoc.meteorError) {
+		error = new Meteor.Error();
+		delete errorDoc.meteorError;
+	} else {
+		error = new Error();
+	}
+
+	Object.getOwnPropertyNames(errorDoc).forEach((key) => {
+		error[key] = errorDoc[key];
+	});
+	return error;
+};
+
+OAuth._retrievePendingCredential = function(key, ...args) {
+	const credentialSecret = args.length > 0 && args[0] !== undefined ? args[0] : null;
+	check(key, String);
+
+	const pendingCredential = OAuth._pendingCredentials.findOne({
+		key,
+		credentialSecret,
+	});
+
+	if (!pendingCredential) {
+		return;
+	}
+
+	if (pendingCredential.credential?.serviceData?._OAuthCustom && !pendingCredential.credential.error) {
+		// Work-around to make the credentials reusable for 2FA
+		const future = new Date();
+		future.setMinutes(future.getMinutes() + 2);
+
+		OAuth._pendingCredentials.update({
+			_id: pendingCredential._id,
+		}, {
+			$set: {
+				_createdAt: future,
```

Comment:
according to meteor code (https://github.com/meteor/meteor/blob/1e7e56eec8414093cd0c1c70750b894069fc972a/packages/oauth/pending_credentials.js#L30) the field is actually called `createdAt`
```suggestion
				createdAt: future,
```